src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/14.diff.sql

```diff
@@ -0,0 +1,537 @@
+
+/*************************************************************
+    Resource change data table
+**************************************************************/
+
+IF NOT EXISTS (SELECT 1 FROM sys.tables WHERE name = 'ResourceChangeData')
+BEGIN
+     CREATE TABLE dbo.ResourceChangeData
+     (
+         Id bigint IDENTITY(1,1) NOT NULL,
+         Timestamp datetime2(7) NOT NULL CONSTRAINT DF_ResourceChangeData_Timestamp DEFAULT sysutcdatetime(),
+         ResourceId varchar(64) NOT NULL,
+         ResourceTypeId smallint NOT NULL,
+         ResourceVersion int NOT NULL,
+         ResourceChangeTypeId tinyint NOT NULL,
+        CONSTRAINT PK_ResourceChangeData PRIMARY KEY CLUSTERED (Id)
+     )
+     ON [PRIMARY]
+END
+GO
+
+/*************************************************************
+    Resource change type table
+**************************************************************/
+
+IF NOT EXISTS (SELECT 1 FROM sys.tables WHERE name = 'ResourceChangeType')
+BEGIN
+     CREATE TABLE dbo.ResourceChangeType
+     (
+         ResourceChangeTypeId tinyint NOT NULL,
+         Name nvarchar(50) NOT NULL,
+         CONSTRAINT PK_ResourceChangeType PRIMARY KEY CLUSTERED (ResourceChangeTypeId),
+         CONSTRAINT UQ_ResourceChangeType_Name UNIQUE NONCLUSTERED (Name)
+     ) 
+     ON [PRIMARY]
+END
+GO
+
+IF (EXISTS (SELECT 1 FROM sys.tables WHERE name = 'ResourceChangeType') AND
+     NOT EXISTS (SELECT 1 FROM dbo.ResourceChangeType WHERE ResourceChangeTypeId = 0))
+BEGIN
+     INSERT dbo.ResourceChangeType (ResourceChangeTypeId, Name) VALUES (0, N'Creation')
+END
+GO
+
+IF (EXISTS (SELECT 1 FROM sys.tables WHERE name = 'ResourceChangeType') AND
+     NOT EXISTS (SELECT 1 FROM dbo.ResourceChangeType WHERE ResourceChangeTypeId = 1))
+BEGIN
+     INSERT dbo.ResourceChangeType (ResourceChangeTypeId, Name) VALUES (1, N'Update')
+END
+GO
+
+IF (EXISTS (SELECT 1 FROM sys.tables WHERE name = 'ResourceChangeType') AND
+     NOT EXISTS (SELECT 1 FROM dbo.ResourceChangeType WHERE ResourceChangeTypeId = 2))
+BEGIN
+     INSERT dbo.ResourceChangeType (ResourceChangeTypeId, Name) VALUES (2, N'Deletion')
+END
+GO
+
+/*************************************************************
+    Stored procedures for capturing and fetching resource changes
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     CaptureResourceChanges
+--
+-- DESCRIPTION
+--     Inserts resource change data
+--
+-- PARAMETERS
+--     @isDeleted
+--         * Whether this resource marks the resource as deleted.
+--     @version
+--         * The version of the resource being written
+--     @resourceId
+--         * The resource ID
+--     @resourceTypeId
+--         * The ID of the resource type
+--
+-- RETURN VALUE
+--     It does not return a value.
+--
+CREATE OR ALTER PROCEDURE dbo.CaptureResourceChanges
+    @isDeleted bit, 
+    @version int, 
+    @resourceId varchar(64),
+    @resourceTypeId smallint
+AS
+BEGIN
```

Comment:
Nit: it might be nice to add a comment explaining that this procedure is intended to be called from the `UpsertResource_4` stored procedure (which is why it doesn't begin a new transaction)

Reply:
Added a comment.